Obey ABSL_DEFAULT_LINKOPTS for all cc_library targets

jwnimmer-tri opened the pull request:
A few targets were missing `linkopts = ...` and so were not obeying the project-wide default settings.

In my downstream project that uses Abseil, I customize the default `linkopts` by patching `configure_copts.bzl`, and having the options missing in a few places causes linker errors.

---

I've confirmed that this patch is comprehensive by running `bazel query 'kind(cc_library, //...) except attr(linkopts, needle, //...)'` after adding this patch:

```patch
--- a/absl/copts/configure_copts.bzl
+++ b/absl/copts/configure_copts.bzl
@@ -35,10 +35,7 @@ ABSL_TEST_COPTS = ABSL_DEFAULT_COPTS + select({
     "//conditions:default": ABSL_GCC_TEST_FLAGS,
 })
 
-ABSL_DEFAULT_LINKOPTS = select({
-    "//absl:msvc_compiler": ABSL_MSVC_LINKOPTS,
-    "//conditions:default": [],
-})
+ABSL_DEFAULT_LINKOPTS = ["needle"]
 
 # ABSL_RANDOM_RANDEN_COPTS blaze copts flags which are required by each
 # environment to build an accelerated RandenHwAes library.
```

derekmauro reviewed (changes requested):
Can you revert that changes to `absl/time/internal/cctz/BUILD.bazel`? They need to be handled through a separate process since that code is imported from the CCTZ project. I'm happy to handle that part for you.

jwnimmer-tri committed:
Obey ABSL_DEFAULT_LINKOPTS for all cc_library targets

A few targets were missing `linkopts = ...` and so were not obeying
the project-wide default settings.

Omit any changes to cctz for now, because it's vendored from another project.

jwnimmer-tri commented:
Thanks for the feedback -- done and force-pushed now.

I re-confirmed that this still catches all linkopts vs the `bazel query` manual testing, other than `cctz`.  (No new libraries snuck in.)

jwnimmer-tri commented:
> Can you revert that changes to absl/time/internal/cctz/BUILD.bazel? They need to be handled through a separate process since that code is imported from the CCTZ project.

FYI @derekmauro I've proposed this follow-up change at https://github.com/google/cctz/pull/305 now.

derekmauro commented:
As of https://github.com/abseil/abseil-cpp/commit/60d1ce9911e6b32ea0c5a2fcfc120f9b392d5643 the CCTZ build files changes are in.

jwnimmer-tri commented:
Perfect, thanks much!
